webapp/src/utils/validationUtils.ts

```diff
@@ -0,0 +1,80 @@
+import { t } from "i18next";
+
+interface ValidationOptions {
+  pattern?: RegExp;
+  existingEntries?: string[];
+  excludedEntries?: string[];
+  isCaseSensitive?: boolean;
+  min?: number;
+  max?: number;
+}
+
+/**
+ * Validates a single string value against specified options.
+ *
+ * @param {string} value - The string to validate, leading/trailing spaces will be trimmed.
+ * @param {ValidationOptions} [options] - Customizable options for validation.
+ *    - `pattern`: RegExp for matching the string. Default to alphanumeric, spaces and "&()_-" pattern.
+ *    - `existingEntries`: Array of strings to check for duplicates. Optional, case-insensitive by default.
+ *    - `excludedEntries`: Array of strings that are explicitly not allowed.
+ *    - `isCaseSensitive`: Default to case-insensitive comparison with `existingEntries`. e.g: "A" and "a" are considered the same.
+ *    - `min`: Minimum length required. Defaults to 0.
+ *    - `max`: Maximum allowed length. Defaults to 50.
+ * @returns {string | true} - True if validation is successful, or a localized error message if it fails.
+ */
+export const validateString = (
+  value: string,
+  options?: ValidationOptions,
+): string | true => {
+  const {
+    pattern,
+    existingEntries = [],
+    excludedEntries = [],
+    isCaseSensitive,
+    min,
+    max,
+  } = {
+    pattern: /^[a-zA-Z0-9_\-() &]+$/,
+    isCaseSensitive: false,
+    min: 0,
+    max: 50,
```

Comment:
there is no particular reason for the max of 50, it seems a good tradeoff because the default 255 VARCHAR is too long

for the special chars its a common pattern used in the app but if its a problem I can consider adding a boolean to configure this like `allowSpecialChars`

Reply:
50 is maybe too small